core/domain/question_jobs_one_off.py

```diff
@@ -91,3 +92,48 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class QuestionsMathRteAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks for existence of math components in the questions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [question_models.QuestionModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        question = question_services.get_question_by_id(item.id)
+        html_string = ''.join(
+            question.question_state_data.
+            get_all_html_content_strings())
+        list_of_latex_strings_without_svg = (
+            html_validation_service.get_latex_strings_without_svg_from_html(
+                html_string))
+        if len(list_of_latex_strings_without_svg) > 0:
+            yield (
+                'Found question with Latex having no SVG.',
+                (item.id, list_of_latex_strings_without_svg))
+
+    @staticmethod
+    def reduce(key, values):
+        final_values = [ast.literal_eval(value) for value in values]
+        total_number_of_latex_strings_without_svg = 0
+        questions_latex_strings_count = []
+        for question_id, latex_strings in final_values:
+            total_number_of_latex_strings_without_svg += len(latex_strings)
+            questions_latex_strings_count.append({
+                'question_id': question_id,
+                'latex_strings_without_svg': latex_strings
+            })
+        yield (
+            'Overall result.', {
```

Comment:
Two general notes:

- For the error cases, could you output details of which question ID, etc. they come from? This is important because maybe we can just recreate them (while there aren't that many).
- Is it possible to output (separately) the cases where there are math components, but they do have the svgs? That will give some confidence that the job is working correctly. But if that's not straightforward, don't worry about it.

Reply:
done